installer.sh

```diff
@@ -59,27 +59,38 @@ confirm_force_install () {
     esac
 }
 
-
+NEED_BUILD_TOOLS=0
 # Which package management system are we using?
 if [[ -n "$(command -v dnf)" ]]; then
     PACKAGE_MGR=$(command -v dnf)
     PYTHON_PREIN="python3 python3-devel python3-setuptools git wget patch"
-    PYTHON_DEPS="python3-pip gcc gcc-c++ openssl-devel swig python3-pyyaml python3-m2crypto  python3-cryptography python3-tornado python3-simplejson python3-requests yaml-cpp-devel procps-ng"
-    PYTHON_PIPS="tornado pyzmq"
+    PYTHON_DEPS="python3-pip gcc gcc-c++ openssl-devel swig python3-pyyaml python3-m2crypto  python3-zmq python3-cryptography python3-tornado python3-simplejson python3-requests yaml-cpp-devel procps-ng"
     BUILD_TOOLS="openssl-devel libtool make automake pkg-config m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen"
+    GOPKG="golang"
+    # if fedora 30 or greater, then using TPM2 tool packages
+    if [[ -f "/etc/fedora-release" && $(cat /etc/fedora-release | awk '{print $3}') -gt 29 ]] ; then
+    	TPM2_TOOLS_PKGS="tpm2-tools tpm2-tss tpm2-abrmd"
+    	NEED_BUILD_TOOLS=0
+    	HAS_GO_PKG=1
+    else
+    	NEED_BUILD_TOOLS=1
+    fi
 elif [[ -n "$(command -v yum)" ]]; then
     PACKAGE_MGR=$(command -v yum)
     $PACKAGE_MGR -y install epel-release
     PYTHON_PREIN="python36 python36-devel python36-setuptools python36-pip git wget patch openssl"
-    PYTHON_DEPS="gcc gcc-c++ openssl-devel swig python36-PyYAML python36-tornado python3-cryptography python36-simplejson python36-requests yaml-cpp-devel"
-    PYTHON_PIPS="tornado pyzmq m2crypto"
-    BUILD_TOOLS="openssl-devel libtool make automake m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen"
+    PYTHON_DEPS="gcc gcc-c++ openssl-devel swig python36-PyYAML python36-tornado python36-simplejson python36-requests yaml-cpp-devel"
```

Comment:
Can we keep the `cryptography` library coming from the repos instead of pip?  

The current script in `master` has a typo -- this should be targeting `python36-cryptography` and not `python3-cryptography`, which is why this is broken on `master` right now.

Reply:
+1

It would be good to avoid pip whenever we can going forward and stick with package managers (no reliance on an internet connection then as some operators run their own internal package repositories, especially in LABs).  Some of the big distributiors (rhel, opensuse etc), will also only support stuff they have packaged themselves (especially for security SLAs).

Reply:
the only one that's not packaged is m2crypto for py36